packages/provider/wrapper.js

```diff
@@ -32,6 +64,13 @@ module.exports = {
       const originalSendAsync = provider.sendAsync.bind(provider);
       provider.sendAsync = this.send(originalSendAsync, preHook, postHook);
     }
+
+    // EIP-1193 provider wrapping
+    if (provider.request === undefined && provider.send !== undefined) {
+      provider.request = make1193ProviderRequest(provider);
+    }
```

Comment:
@gnidan , @cds-amal and I thought of making a separate function for the 1193 type.